wordpress-mobile/WordPress-iOS
# Add navigation to all domains screen

Fixes #22248 

## Description

This PR adds the function that navigates to All Domains screen. Due to the fact that app-site-association file is needed to be updated for the new universal links to work, we'll need to manually call the function to test the functionality.

## Testing Steps
Call `RootViewCoordinator.sharedPresenter.navigateToAllDomains()` from anywhere within the codebase to trigger the navigation to Me -> All Domains.

Question to Reviewer:
Should the navigation open the `Me` Screen first or should it directly open the all domains screen on wherever the app is currently. I opted for the former as I thought this would also highlight the location of it in case they want to access it again.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


This PR Does not impact UI

UI Changes testing checklist:
